data/datasets/poetry_instruction/prepare.py

```diff
@@ -0,0 +1,151 @@
+import json
+import os
+import random
+
+import pandas as pd
+
+# Read the CSV file into a pandas DataFrame. The CSV file was downloaded from Kaggle, and the source is linked in the README.
+
+CURRENT_DIR = os.path.dirname(__file__)  # Gets directory path of the current python module
```

Comment:
mabye you can also add a KAGGLE dataset loader to unpack this (I can give you an example code for this)